app/src/org/commcare/android/view/c3/DataConfiguration.java

```diff
@@ -0,0 +1,349 @@
+            mColumns.put(xValues);
+            mColumns.put(yValues);
+            if (mData.getType().equals(Graph.TYPE_BUBBLE)) {
+                radii.put(yID, rValues);
+                maxRadii.put(yID, maxRadius);
+            }
+
+            // Set series name for legend
+            String name = s.getConfiguration("name", "");
+            if (name != null) {
+                mNames.put(yID, name);
+            }
+
+            // Set series color
+            // TODO: Handle transparency (and test on bubble graphs)
+            String color = s.getConfiguration("line-color", "#ff000000");
+            if (color.length() == "#aarrggbb".length()) {
+                color = "#" + color.substring(3);
+            }
+            mColors.put(yID, color);
+
+            // Associate values with either primary or secondary y axis
+            mAxes.put(yID, Boolean.valueOf(s.getConfiguration("secondary-y", "false")).equals(Boolean.TRUE) ? "y2" : "y");
+
+            seriesIndex++;
+        }
+
+        // For bar charts, force the x axis min and max so
+        // bars are spaced nicely and set up bar labels
+        if (mData.getType().equals(Graph.TYPE_BAR)) {
+            mData.setConfiguration("x-min", "0.5");
+            mData.setConfiguration("x-max", String.valueOf(barCount + 0.5));
+            barLabels.put("");
+            mVariables.put("barLabels", barLabels.toString());
+        }
+
+        // Add fake series to force ticks to be based on min and max
+        addBoundaries();
+
+        // Add fake series for annotations
+        addAnnotations();
+
+        // Finally, apply all data to main configuration
+        mConfiguration.put("axes", mAxes);
+        mConfiguration.put("colors", mColors);
+        mConfiguration.put("columns", mColumns);
+        mConfiguration.put("names", mNames);
+        mConfiguration.put("types", mTypes);
+        mConfiguration.put("xs", mXs);
+
+        // Set up stacked bar graph if needed
+        if (mData.getType().equals(Graph.TYPE_BAR)
+                && Boolean.valueOf(mData.getConfiguration("stack", "false")).equals(Boolean.TRUE)) {
+            JSONArray inner = new JSONArray();
+            for (Iterator<String> i = mTypes.keys(); i.hasNext();) {
+                String key = i.next();
+                if (mTypes.get(key).equals("bar")) {
+                    inner.put(key);
+                }
+            }
+            JSONArray outer = new JSONArray();
+            outer.put(inner);
+            mConfiguration.put("groups", outer);
+        }
+
+        // C3 doesn't support bubble charts well, so radius data gets set up
+        // in separate variables rather than as part of the configuration
+        mVariables.put("radii", radii.toString());
+        mVariables.put("maxRadii", maxRadii.toString());
+    }
+
+    /**
+     * Add annotations, by creating a fake series with data labels turned on.
+     * @throws JSONException
+     * @throws InvalidStateException

```

Comment:
`@throws` annotations that don't include descriptions don't offer additional info over the method signature so it would be nice to remove them